cpp/src/IceGrid/AdapterCache.cpp

```diff
@@ -212,38 +195,44 @@ GetAdapterInfoResult::get()
     return _adapters;
 }
 
-AdapterCache::AdapterCache(const Ice::CommunicatorPtr& communicator) : _communicator(communicator)
+AdapterCache::AdapterCache(const shared_ptr<Ice::Communicator>& communicator) : _communicator(communicator)
 {
 }
 
 void
-AdapterCache::addServerAdapter(const AdapterDescriptor& desc, const ServerEntryPtr& server, const string& app)
+AdapterCache::addServerAdapter(const AdapterDescriptor& desc, const shared_ptr<ServerEntry>& server, const string& app)
 {
-    Lock sync(*this);
+    lock_guard lock(_mutex);
     if(getImpl(desc.id))
     {
         Ice::Error out(_communicator->getLogger());
         out << "can't add duplicate adapter `" << desc.id << "'";
         return;
     }
 
-    istringstream is(desc.priority);
```

Comment:
Why not keep istringstream here?

Reply:
I think parsing the string with [stoi](https://en.cppreference.com/w/cpp/string/basic_string/stol) is cleaner (and leaner) here.